### PR TITLE
ROX-9855: Enable image signature verification feature flag by default.

### PR DESCRIPTION
**Pull request by dhaus67**
## Description

Enable the `ROX_VERIFY_IMAGE_SIGNATURE` feature flag by default.
Remove paths where it was used to conditionally i.e. skip tests / create integrations.

For now, not removing all the code paths, in a case any blocking issues occur during the release testing, we will be able to roll back.

## Testing Performed
- CI (`race-condition-tests` should execute `ImageSignatureVerificationTest` and be successful)

### CODE REVIEW

File: pkg/images/enricher/enricher_impl.go [hunk 1]
```diff
@@ -575,6 +575,10 @@ func (e *enricherImpl) enrichWithSignature(ctx context.Context, enrichmentContex
 
 	matchingRegistries, err := getMatchingRegistries(registrySet.GetAll(), img)
 	if err != nil {
+		// Do not return an error for internal images when no integration is found.
```
Comment (ivan-degtiarenko) [inline pkg/images/enricher/enricher_impl.go:578]: Why is this change needed?

File: pkg/images/enricher/enricher_impl_test.go [hunk 2]
```diff
@@ -623,7 +634,9 @@ func TestNoMatchingRegistryIntegration(t *testing.T) {
 	img := &storage.Image{Id: "id", Name: &storage.ImageName{Registry: "reg"}}
 	results, err := enricherImpl.EnrichImage(emptyCtx, EnrichmentContext{}, img)
 	assert.Error(t, err)
-	expectedErrMsg := "image enrichment error: error getting metadata for image:  error: no matching image registries found: please add an image integration for reg"
+	expectedErrMsg := "image enrichment errors: [error getting metadata for image:  error: no matching image " +
```
Comment (ivan-degtiarenko) [inline pkg/images/enricher/enricher_impl_test.go:637]: why did error change?

**Review by ivan-degtiarenko — COMMENTED** [review]
`Verify Image Registry+Scanner Integrations` e2e test is failing, could it be due to our change?

Also, should we add changelog line here as well?

**Comment by dhaus67** [comment]
> Verify Image Registry+Scanner Integrations e2e test is failing, could it be due to our change?

There exists a known flake, [ROX-9859](https://issues.redhat.com/browse/ROX-9859) which has been there prior to any changes we did related to image signing, so I wouldn't see this as a blocker.

File: pkg/images/enricher/enricher_impl.go (hunk 1 above)
Comment (dhaus67) [reply pkg/images/enricher/enricher_impl.go:578]: When looking at the failing test, we expect the enricher to _not_ return an error when no registries are available and `enrichmentContext.Internal` is given. Had to fix this within the `enrichWithSignature`.

File: pkg/images/enricher/enricher_impl_test.go (hunk 2 above)
Comment (dhaus67) [reply pkg/images/enricher/enricher_impl_test.go:637]: We don't return on the first error occurred within the enrichment flow, but rather are creating an error list.
Previously, only `enrichWithMetadata` was accessing registry, now the `enrichWithSignature` step will also access registry information, thus the second error in the list since it's enabled now by default.

File: CHANGELOG.md [hunk 3]
```diff
@@ -19,6 +19,9 @@ Please avoid adding duplicate information across this changelog and JIRA/doc inp
 - Fixed permissions checks in the UI that prevented users with certain limited permissions from creating report configurations.
 - ROX-9946: Fixed default permissions for the default Vuln Reporter role to exclude the modify permission on notifiers, since it is not needed for report creation.
 - Added AllowPrivilegeEscalation as a new policy criteria.
+- Image signatures are now supported throughout the system, specifically cosign signatures. A new integration type for
```
Comment (rukletsov) [inline CHANGELOG.md:22]: I'm not sure we need to call out new feature in the changelog, they shall be added automatically from jira and by product. As long as there are no changes or deprecations to the API, it's okay to leave out.

Comment (dhaus67) [reply CHANGELOG.md]: Ah, that explains why I didn't see any examples 🥲

**Review by ivan-degtiarenko — APPROVED** [review]
LGTM
